src/resolveParams.ts

```diff
@@ -0,0 +1,31 @@
+import {DocumentsPaneQueryParams, DocumentVersionsCollection} from './types'
+import delve from 'dlv'
+
+interface ResolveParamsOptions {
+  params?: DocumentsPaneQueryParams
+  document: DocumentVersionsCollection
+  useDraft: boolean
+}
+
+type ResolveParamsReturn = null | {[key: string]: string}
+
+export default function resolveParams(options: ResolveParamsOptions): ResolveParamsReturn {
+  const {params, document, useDraft} = options
+
+  if (typeof params == 'function') {
+    return params({document})
+  }
+
+  // legacy useDraft behaviour
+  const doc = useDraft ? document.displayed : document.published
+  const {_rev} = doc ?? {}
+
+  // return null so that we can show the warning when the document hasn't been published yet
```

Comment:
Question from #10 - should the query params resolver return `null` or `undefined` if the parameters can't be resolved?

Reply:
Your note is probably correct there that idea is incorrect anyway since a draft will have a `_rev` key anyway. Perhaps we just remove that logic.

Reply:
Well, there are still cases where you won't be able to resolve parameters. Particularly, the `_id` is not set (even in `displayed`) until the first mutation. So I think we still need a way for the default and custom resolvers to indicate that props couldn't be resolved, so the query should not be attempted. Otherwise you get a query error due to the props not being defined, which doesn't seem like good UX.

Maybe what we should do instead is check if any props values are `undefined`, whether returned from the custom or default resolver, since that will cause a query parse error. I think I'll make that change and push it up. It will simplify the custom resolvers.